### PR TITLE
media-video/vlc: Prefer qt5 to qt4 if both enabled

### PR DESCRIPTION
**Pull request by a17r**
Drops another unnecessary REQUIRED_USE, but keeping Qt4 here for alleged scaling issues with Qt5...

Also fix build with >=Qt-5.7.0 for vlc-2.2.4 to be able to make progress with Qt-5.7 unmasking.

@gentoo/video @SDNick484

### CODE REVIEW

**Comment by Amynka** [comment]
@gentoo/proxy-maint

File: media-video/vlc/vlc-2.2.1-r1.ebuild [hunk 1]
```diff
@@ -289,14 +288,16 @@ src_prepare() {
 
 	# If qtchooser is installed, it may break the build, because moc,rcc and uic binaries for wrong qt version may be used.
 	# Setting QT_SELECT environment variable will enforce correct binaries.
-	if use qt4; then
-		export QT_SELECT=qt4
-	elif use qt5; then
+	if use qt5; then
 		export QT_SELECT=qt5
+	else
```
Comment (Pesa) [inline media-video/vlc/vlc-2.2.1-r1.ebuild]: why not `elif use qt4; ...`?

File: media-video/vlc/files/vlc-2.2.4-cxx11.patch [hunk 2]
```diff
@@ -0,0 +1,11 @@
+--- vlc-2.2.4/configure.ac.orig	2016-09-21 07:10:58.885508665 +0200
++++ vlc-2.2.4/configure.ac	2016-09-21 07:19:17.835725004 +0200
+@@ -3746,7 +3746,7 @@
+     PKG_CHECK_MODULES([QT], [Qt5Core >= 5.1.0 Qt5Widgets Qt5Gui], [
+       PKG_CHECK_MODULES([QTX11], [Qt5X11Extras], [
+           VLC_ADD_LIBS([qt4],[${QTX11_LIBS}])
+-          VLC_ADD_CXXFLAGS([qt4],[${QTX11_CFLAGS} -DQT5_HAS_X11])
++          VLC_ADD_CXXFLAGS([qt4],[${QTX11_CFLAGS} -DQT5_HAS_X11 -std=c++11])
```
Comment (Pesa) [inline media-video/vlc/files/vlc-2.2.4-cxx11.patch]: I think this should be `c++0x` or `gnu++0x`

Comment (a17r) [inline media-video/vlc/files/vlc-2.2.4-cxx11.patch]: sure, works for me as well!

**Comment by kensington** [comment]
Thanks, picked as 66b2ab8f5f64ca3e7a4835477620c25d736d6bad...6ffd3e6806dc84eabb4ca910038a4628304b8db2.

**Pull request closed without merging**